index.js

```diff
@@ -14,53 +13,65 @@ var globals = require('./globals');
 
 var http = require('http');
 var https = require('https');
-var selfCaller = function(path, req, res, cb, url) {
+var selfCaller = function (path, req, res, cb, url) {
   var url = req.get('host').split(':');
   var port = url[1];
   url = url[0];
   var prot = req.protocol === 'https' ? https : http;
   var opts = {
-    hostname : url,
-    method : 'GET',
-    path : path,
-    port : port || 80
+    hostname: url,
+    method: 'GET',
+    path: path,
+    port: port || 80,
+    headers: {
+      'User-Agent':
+        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36',
+    },
   };
-  var rq = prot.request(opts, function(r){
-    r.on('data', (d) => {cb(d.toString(), req, res, r.headers)});
-    r.on('error', () => {res.status(500).type('txt').send('SERVER ERROR')});
+  var rq = prot.request(opts, function (r) {
+    r.on('data', (d) => {
+      cb(d.toString(), req, res, r.headers);
+    });
+    r.on('error', () => {
+      res.status(500).type('txt').send('SERVER ERROR');
+    });
     r.resume();
   });
   rq.end();
 };
 
-var enableCORS = function(req, res, next) {
+var enableCORS = function (req, res, next) {
   if (!process.env.DISABLE_XORIGIN) {
-    var allowedOrigins = ['https://narrow-plane.gomix.me', 'https://www.freecodecamp.com'];
+    var allowedOrigins = [
+      'https://narrow-plane.gomix.me',
+      'https://www.freecodecamp.com',
+    ];
     var origin = req.headers.origin;
-    if(!process.env.XORIG_RESTRICT || allowedOrigins.indexOf(origin) > -1) {
-       res.set({
+    if (!process.env.XORIG_RESTRICT || allowedOrigins.indexOf(origin) > -1) {
+      res.set({
         'Access-Control-Allow-Origin': origin,
-        "Access-Control-Allow-Headers" : "Origin, X-Requested-With, Content-Type, Accept"
-       });
+        'Access-Control-Allow-Headers':
+          'Origin, X-Requested-With, Content-Type, Accept',
+      });
     }
   }
   next();
 };
 
 function setupBackgroundApp(app, myApp, dirname) {
   app.use(enableCORS);
-  app.get('/_api/hello-console', function(req, res) {
-    res.json({passed: globals.userPassedConsoleChallenge});
+  app.get('/_api/hello-console', function (req, res) {
+    res.json({ passed: globals.userPassedConsoleChallenge });
   });
 
-  app.get('/_api/json', function(req, res) {
+  app.get('/_api/json', function (req, res) {
     var msgStyle = process.env.MESSAGE_STYLE;
     process.env.MESSAGE_STYLE = undefined;
-    selfCaller('/json', req, res, function(lowerCase, req, res) {
+    selfCaller('/json', req, res, function (lowerCase, req, res) {
       process.env.MESSAGE_STYLE = msgStyle;
       try {
         lowerCase = JSON.parse(lowerCase);
-      } catch(e) {
+      } catch (e) {
         console.log(e);
         process.env.MESSAGE_STYLE = msgStyle;
         next(e);
```

Comment:
I am not entirely familiar with the working of this function, but as mentioned in another issue, `next()` either needs to be passed to be used here, or this line needs to be removed.

Reply:
I guess it's fine to fix it in this PR. Done.

Reply:
It's fine, I just won't squash the commits.

Reply:
So, this is kinda weird, because it seems to be passing around stuff that's already available via closure.  I think it's fine, just confusingly written.